### ISSUE TITLE
Angular 6: "Cannot read property 'CLASS_TO_PLAIN' of undefined" when using the `buildOptimizer`

### ISSUE THREAD

**Issue by realmar** [typestack/class-transformer#147]
Hey all!

I'm using `class-transformer` in my angular app. I recently upgraded the project to Angular 6 and `class-transformer` started throwing the following exceptions when transforming from class to plain and from plain to class: 

```
// class to plain
Cannot read property 'CLASS_TO_PLAIN' of undefined

// plain to class
Cannot read property 'PLAIN_TO_CLASS' of undefined
```

I noticed that _those issues only occur when enabling the `buildOptimizer: true` in `angular.json`._ I would appreciate any help on the matter!

**Comment by realmar** [comment]
This is an issue with the angular build-optimizer and has been discussed [here](https://github.com/angular/angular-cli/issues/7613). As per [this](https://github.com/angular/angular-cli/issues/7613#issuecomment-330849089) comment, the issue should be fixed. However, it seems like there is a regression in the full release of v6 which causes the enums to be stripped out again: 

- https://github.com/angular/angular-cli/issues/10655
- https://github.com/angular/angular-cli/issues/10658

A workaround is described [here](https://github.com/angular/angular-cli/issues/7613#issuecomment-328780936)

I have cloned `class-transformer` and have fixed the issue locally by just using `string`s for the `TransformationType` enum: (I also moved this `enum` to a separate file and removed the duplicate definition of it in `index.ts`)

```typescript
export enum TransformationType {
    PLAIN_TO_CLASS = "0",
    CLASS_TO_PLAIN = "1",
    CLASS_TO_CLASS = "2"
}
```

If you would like I can open a PR with the changes, however I don't know if that would be fitting as the described behavior is a bug in angular and not in `class-transformer`.

**Comment by realmar** [comment]
This issue has now been fixed in [v6.0.1 of the angular devkit](https://github.com/angular/devkit/releases/tag/v6.0.1) (https://github.com/angular/devkit/commit/959826bda881e97a4f8bbe0a6ef77e38669c9076). Closing this issue.

**Comment by NoNameProvided** [comment]
Thanks for the clear explanation for the future readers.
